MetaMask/metamask-extension
# feat: create Developer Options Setting Page

## **Description**

The "Developer Options" setting page will provide developers with additional controls and customization to assist with development and testing. This feature will enable developers to reset app states easily, toggle certain features on/off, and update configurations without needing to reinstall the app, accelerating developer velocity.

To enable, add `ENABLE_SETTINGS_PAGE_DEV_OPTIONS=true` in .metamaskrc

Developer Options Supported:
- Reset Announcements/Notifications for What's New modal
- Reset Onboarding states and redirects to second onboarding page


## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/23592
Blocked By: https://github.com/MetaMask/core/pull/4088

## **Manual testing steps**

1. `yarn install` to install the latest announcement-controller changes
2. go to `.metamaskrc`
3. paste `ENABLE_SETTINGS_PAGE_DEV_OPTIONS=true`
4. `yarn start`

Test using test-dapp
1. open test-dapp
2. test with PPOM buttons

---

1. See chrome-extension://hbljfohiafgaaaabejngpgolnboohpaf/home.html#settings/developer-options

## **Screenshots/Recordings**

#### New UI

<img width="480" alt="Screenshot 2023-12-21 at 2 33 45 PM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/6d6ebd6c-8564-4d4e-8dcf-d9197ad11e6b">

![Screenshot 2023-12-22 at 1 58 28 PM](https://github.com/MetaMask/metamask-extension/assets/20778143/cfec8e2e-5cf8-43ed-9096-af22901e4bd5)

![Screenshot 2023-12-22 at 1 59 40 PM](https://github.com/MetaMask/metamask-extension/assets/20778143/70a11c03-735c-4f98-93e4-d8532c5aaea4)

#### Announcements 

![Screenshot 2024-03-17 at 8 13 41 PM](https://github.com/MetaMask/metamask-extension/assets/20778143/be91edfa-ade3-4a37-b628-3c8c62b0d27c)

#### Onboarding

![Screenshot 2024-03-17 at 8 13 56 PM](https://github.com/MetaMask/metamask-extension/assets/20778143/ae36304a-db40-4a7a-bbf2-e979167ecf4c)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
